### ISSUE TITLE
White_light error processing NIRISS SOSS exposure

### ISSUE THREAD

**Issue by hbushouse** [STScI-JWST/jwst#2148]
A NIRISS SOSS exposure contained in the DMS test suite is failing in ``calwebb_tso3`` processing in the ``white_light`` step, with the following traceback:
```
2018-06-11 12:44:11,331 - stpipe.Tso3Pipeline.extract_1d - WARNING - Using RELSENS, 1550 elements were extrapolated; these values will be set to 1.
2018-06-11 12:44:11,390 - stpipe.Tso3Pipeline.extract_1d - INFO - All 369 integrations done
2018-06-11 12:44:11,390 - stpipe.Tso3Pipeline.extract_1d - WARNING - INTSTART not found; assuming a value of 1.
2018-06-11 12:44:11,390 - stpipe.Tso3Pipeline.extract_1d - WARNING - INTEND not found; assuming a value of 1639.
2018-06-11 12:44:11,390 - stpipe.Tso3Pipeline.extract_1d - WARNING - There is no INT_TIMES table in the input file.
2018-06-11 12:44:11,391 - stpipe.Tso3Pipeline.extract_1d - INFO - Step extract_1d done
2018-06-11 12:44:15,286 - stpipe.Tso3Pipeline - INFO - Performing white-light photometry...
2018-06-11 12:44:15,940 - stpipe.Tso3Pipeline.white_light - INFO - Step white_light running with args (<MultiSpecModel from jw10003001001_03101_00001-seg003_nis_calints.fits>,).
----------------------------------------------------------------------
ERROR RUNNING STEP 'Tso3Pipeline':
    list index out of range
Traceback (most recent call last):
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.7a0.dev3-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/cmdline.py", line 297, in step_from_cmdline
    step.run(*positional)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.7a0.dev3-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 398, in run
    step_result = self.process(*args)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.7a0.dev3-py3.5-macosx-10.7-x86_64.egg/jwst/pipeline/calwebb_tso3.py", line 135, in process
    phot_result_list.append(self.white_light(result))
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.7a0.dev3-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 398, in run
    step_result = self.process(*args)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.7a0.dev3-py3.5-macosx-10.7-x86_64.egg/jwst/white_light/white_light_step.py", line 25, in process
    result = white_light(input_model)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.7a0.dev3-py3.5-macosx-10.7-x86_64.egg/jwst/white_light/white_light.py", line 23, in white_light
    fluxsums.append(input.spec[i].spec_table['FLUX'].sum())
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.7a0.dev3-py3.5-macosx-10.7-x86_64.egg/jwst/datamodels/properties.py", line 293, in __getitem__
    return _make_node(self._name, self._instance[i], schema, self._ctx)
IndexError: list index out of range

----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.7a0.dev3-py3.5-macosx-10.7-x86_64.egg/EGG-INFO/scripts/strun", line 14, in <module>
    step = Step.from_cmdline(sys.argv[1:])
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.7a0.dev3-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 162, in from_cmdline
    return cmdline.step_from_cmdline(args)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.7a0.dev3-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/cmdline.py", line 297, in step_from_cmdline
    step.run(*positional)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.7a0.dev3-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 398, in run
    step_result = self.process(*args)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.7a0.dev3-py3.5-macosx-10.7-x86_64.egg/jwst/pipeline/calwebb_tso3.py", line 135, in process
    phot_result_list.append(self.white_light(result))
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.7a0.dev3-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 398, in run
    step_result = self.process(*args)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.7a0.dev3-py3.5-macosx-10.7-x86_64.egg/jwst/white_light/white_light_step.py", line 25, in process
    result = white_light(input_model)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.7a0.dev3-py3.5-macosx-10.7-x86_64.egg/jwst/white_light/white_light.py", line 23, in white_light
    fluxsums.append(input.spec[i].spec_table['FLUX'].sum())
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.7a0.dev3-py3.5-macosx-10.7-x86_64.egg/jwst/datamodels/properties.py", line 293, in __getitem__
    return _make_node(self._name, self._instance[i], schema, self._ctx)
IndexError: list index out of range
```
The input data files are available in /grp/jwst/ssb/bushouse/jwst_data/NIRISS/SOSS/ and are:
jw10003-o001_20180609t084822_tso3_001_asn.json
jw10003001001_03101_00001-seg003_nis_calints.fits

**Comment by hbushouse** [comment]
Yes, this file was segmented. The decision was made a while back to leave the original NINTS value in that keyword for segmented files, so we can't rely on it directly anymore. We need to just look at the shape of the data array (as you suggested). Eventually we may be able to start using INTSTART and INTEND, once they get added to the headers, but even then it's safest to use the actual data array shape. Don't worry about the overlapping times for now.

**Comment by hbushouse** [comment]
Fixed by #2155.
